http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/AwsCrtAsyncHttpClient.java

```diff
@@ -340,9 +371,58 @@ Builder connectionHealthChecksConfiguration(Consumer<ConnectionHealthChecksConfi
                                                         healthChecksConfigurationBuilder);
 
         /**
-         * Configure the maximum amount of time that a connection should be allowed to remain open while idle.
+         * The amount of time to wait when initially establishing a connection before giving up and timing out. The maximum
+         * possible value, in ms, is the value of {@link Integer#MAX_VALUE}, any longer duration will be reduced to the maximum
+         * possible value. If not specified, the connection timeout duration will be set to value defined in
+         * {@link AwsCrtAsyncHttpClient#CRT_SDK_DEFAULT_CONNECTION_TIMEOUT}.
          */
         Builder connectionMaxIdleTime(Duration connectionMaxIdleTime);
+
+        /**
+         * Configure connection socket timeout
+         */
+        Builder connectionTimeout(Duration connectionTimeout);
+
+        /**
+         * Configure whether to enable or disable TCP KeepAlive.
+         * The configuration will be passed to the socket option {@link java.net.SocketOptions#SO_KEEPALIVE}.
+         * <p>
+         * By default, this is disabled.
+         * <p>
+         * When enabled, the actual KeepAlive mechanism is dependent on the Operating System and therefore additional TCP
+         * KeepAlive values (like timeout, number of packets, etc) must be configured via the Operating System (sysctl on
+         * Linux/Mac, and Registry values on Windows).
+         *
+         * To enable, keepAlive interval and timeout must be additionally configured via {@link TcpKeepAliveConfiguration}
+         */
+        Builder tcpKeepAlive(Boolean keepConnectionAlive);
```

Comment:
The documentation doesn't seem to be correct here since it does not correspond to the Java tcp keepalive implementation. I thought we are going to just remove the boolean tcpKeepAlive since we decided to require customers to provide tcpKeepAlive timeouts (Sorry for being unclear)

Reply:
I tried to have it both ways - have the boolean flag matching the Netty interface, AND have the additional configuration flags since they're important.

But you're right that doesn't provide enough value, will take it out.